src/farcasterd/runtime.rs

```diff
@@ -1014,315 +489,11 @@ impl Runtime {
             Request::CheckpointList(checkpointed_pub_offers) => {
                 self.checkpointed_pub_offers = checkpointed_pub_offers.clone();
```

Comment:
since cleaning up redundant state anyway, this `checkpointed_pub_offers` is quite unwieldy. Seems it's only needed for the trade state machine to get the public offer & trade role when fed a swap id.

Lightly suggest moving the public offer & trade role into the event msg passed to `transition_to_restoring_swapd` as well to get rid of this.

Reply:
The important bit about the `checkpointed_pub_offers` is not necessarily the offer and trade role, but rather that it allows us to immediately report back to the user if the checkpoint does not exist. To get rid of this cache, we'd have to introduce further messages and interim states to the restore workflow in order to report back to the user in time if the checkpoint does not exist. This is not an easy change to tackle, so I'd rather defer this for a separate PR.

Reply:
sure, we can leave it for later.

> it allows us to immediately report back to the user if the checkpoint does not exist

some issues with the restore atm:
if the user hasn't called `list checkpoints` yet and goes straight for `restore checkpoint`, the trade state machine throws `No checkpoint found with given swap id, aborting restore.`, which is misleading for anyone not familiar with the service internals.

imo that tradeoff in favour of reporting back to the user immediately is worse, since in order for the response to their `restore checkpoint` request to have meaning, they have to `list checkpoints` first anyway, which should confirm to them already whether or not the checkpoint exists.

For a message workflow that allows immediate feedback, I think doing
```
cli -> databased : RestoreCheckpoint
if checkpoint-exists {databased -> farcasterd : RestoreCheckpoint} else {databased -> cli: CheckpointNotFound; exit}
farcasterd -> swapd : launch
farcasterd -> databased : RestoreCheckpoint
databased -> walletd : Checkpoint
databased -> swapd: Checkpoint
```
should work - I'll try this out later.

Reply:
I'm already working on it, but will not push to this branch. Yes, the flow you showed here should work.